code-challenges.test.js

```diff
@@ -12,41 +12,169 @@
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.153 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 0.63s.
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+// 
+// Pseudocode
+// Input: array of 5 numbers
+// Output: boolean (true or false)
+// Function name: fullHouse
+// Process: take an array of cards as an argument and use .reduce method on the array to create an object called counts that contains the number of times each card appears in the array. The result of the counts variable will be used in  2 additional variables to return if the object values have 3 of a kind (hasThree)or 2 pairs(hasPair) and return true or false if the object has both using the logical AND operator.
+
+describe("fullHouse", () => {
+        it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse(hand1)).toStrictEqual(true)
+        const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        expect(fullHouse(hand2)).toStrictEqual(false)
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        expect(fullHouse(hand3)).toStrictEqual(false)
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        expect(fullHouse(hand4)).toStrictEqual(true)
+        const hand4 = [7, 2, 7, 2, 7]
+        // Expected output: true
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// codedMessage
+//   ✓ takes in a string and returns a string with a coded message (1 ms)
+// wordsThatIncludes
+//   ✓ takes in an array of words and a single letter and returns an array of all the words containing that particular letter (1 ms)
+// fullHouse
+//   ✕ takes in an array of 5 numbers and determines whether or not the array is a full house
+
+// ● fullHouse › takes in an array of 5 numbers and determines whether or not the array is a full house
+
+//   ReferenceError: fullHouse is not defined
 
-// a) Create a test with expect statements using the variable provided.
-
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
 
 // b) Create the function that makes the test pass.
+
+// This problem took a long time to solve. I had many blockers and each blocker led to a ton of tabs opened of google searches. Eventually I landed on the .reduce method and this resource https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce that eventually led me to understand that I can make it create an object after the return on line 159
```

Comment:
This is really great research and work put in! This is my personal favorite approach to this challenge, glad to see you figured this one out!! An important concept that was not mentioned in your pseudo code: when using `.reduce`, it takes a callback function as an argument, with the parameters being an accumulator and the current value in the iteration (in this case, the accumulator is `total` and the current value is `card`. The second argument in `.reduce` is the starting point of the accumulator, so since you put an empty object as the second argument, `total` starts off as an empty object. As you iterate, you are creating keys in the object (using `total[card]`) and setting the value to whatever it currently is and then + 1, or if the key does not exist yet, giving it an initial value of 0 then adding 1 to 0. Love it!!